### PR TITLE
[many ports]switch to vcpkg-cmake / vcpkg-cmake-config part 2

### PR DESCRIPTION
**Pull request by JackBoosY**
Prev: https://github.com/microsoft/vcpkg/pull/29880

### CODE REVIEW

**Comment by JackBoosY** [comment]
depends on https://github.com/microsoft/vcpkg/pull/29916

**Comment by talregev** [comment]
@JackBoosY 
Can you separate the failed ports to different PRs?

**Comment by JackBoosY** [comment]
> @JackBoosY
> Can you separate the failed ports to different PRs?

There is no rush, so we can wait for the dependent PRs to be merged.

**Comment by talregev** [comment]
> > @JackBoosY
> > Can you separate the failed ports to different PRs?
> 
> There is no rush, so we can wait for the dependent PRs to be merged.

Can you put a links for dependent PRs that you are waiting for this PR? Put them in the top comment will help.
Thank you for your hard work!

**Comment by JackBoosY** [comment]
> > > @JackBoosY
> > > Can you separate the failed ports to different PRs?
> > 
> > 
> > There is no rush, so we can wait for the dependent PRs to be merged.
> 
> Can you put a links for dependent PRs that you are waiting for this PR? Put them in the top comment will help. Thank you for your hard work!

Once https://github.com/microsoft/vcpkg/pull/28084 was merged, I will merge this PR to master.

File: ports/kd-soap/portfile.cmake [hunk 1]
```diff
@@ -8,25 +8,25 @@ vcpkg_from_github(
...
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/KDSoap PACKAGE_NAME KDSoap)
 
-file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/tools)
-file(RENAME ${CURRENT_PACKAGES_DIR}/bin/kdwsdl2cpp${VCPKG_TARGET_EXECUTABLE_SUFFIX} ${CURRENT_PACKAGES_DIR}/tools/kdwsdl2cpp${VCPKG_TARGET_EXECUTABLE_SUFFIX})
-file(REMOVE ${CURRENT_PACKAGES_DIR}/debug/bin/kdwsdl2cpp${VCPKG_TARGET_EXECUTABLE_SUFFIX})
+file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools")
+file(RENAME "${CURRENT_PACKAGES_DIR}/bin/kdwsdl2cpp${VCPKG_TARGET_EXECUTABLE_SUFFIX}"
+    "${CURRENT_PACKAGES_DIR}/tools/kdwsdl2cpp${VCPKG_TARGET_EXECUTABLE_SUFFIX}")
+file(REMOVE "${CURRENT_PACKAGES_DIR}/debug/bin/kdwsdl2cpp${VCPKG_TARGET_EXECUTABLE_SUFFIX}")
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
```
Comment (Osyotr) [inline ports/kd-soap/portfile.cmake:25]: While you're at it, might as well fix x64-linux-dynamic build
```suggestion
if(VCPKG_LIBRARY_LINKAGE STREQUAL "static" OR NOT VCPKG_TARGET_IS_WINDOWS)
```

Comment (JackBoosY) [reply ports/kd-soap/portfile.cmake:25]: This PR just remove the deprecated functions and add double quotes to the absolute paths.

Comment (Osyotr) [reply ports/kd-soap/portfile.cmake:25]: Ok, I'll make a PR after this one is merged.

**Comment by MonicaLiu0311** [comment]
@JackBoosY 
Convert this PR to draft since there is no progress. Please ping us if this PR is ready for review again.

**Comment by BillyONeal** [comment]
https://github.com/microsoft/vcpkg/pull/26981 covered a lot of the same ground and as a result some of the changes that are left only add quotes. Do you still think we should have those changes or should we drop the merely quote changes?

File: ports/functions-framework-cpp/portfile.cmake [hunk 2]
```diff
@@ -10,21 +10,21 @@ vcpkg_from_github(
...
     OPTIONS
         -DBUILD_TESTING=OFF
 )
 
 vcpkg_cmake_install(ADD_BIN_TO_PATH)
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
-vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake PACKAGE_NAME share)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake)
```
Comment (BillyONeal) [inline ports/functions-framework-cpp/portfile.cmake:22]: Note that https://github.com/microsoft/vcpkg/pull/26981 added PACKAGE_NAME share which seems wrong

**Review by BillyONeal — DISMISSED** [review]

**Comment by JackBoosY** [comment]
@BillyONeal We still need to switch `vcpkg_fixup_cmake_targets ` to `vcpkg_config_cmake_fixup`.
This PR is preparing for https://github.com/microsoft/vcpkg/pull/29830 and https://github.com/microsoft/vcpkg/pull/30070 so we can continue PR https://github.com/microsoft/vcpkg-tool/pull/675

**Comment by BillyONeal** [comment]
> @BillyONeal We still need to switch `vcpkg_fixup_cmake_targets ` to `vcpkg_config_cmake_fixup`. This PR is preparing for #29830 and #30070 so we can continue PR [microsoft/vcpkg-tool#675](https://github.com/microsoft/vcpkg-tool/pull/675)

I don't see any left:

vcpkg_fixup_cmake_targets search results

**Comment by JackBoosY** [comment]
> > @BillyONeal We still need to switch `vcpkg_fixup_cmake_targets ` to `vcpkg_config_cmake_fixup`. This PR is preparing for #29830 and #30070 so we can continue PR [microsoft/vcpkg-tool#675](https://github.com/microsoft/vcpkg-tool/pull/675)
> 
> I don't see any left:
> 
> vcpkg_fixup_cmake_targets search results

ports/ignition-modularscripts/vcpkg-port-config.cmake

**Comment by JackBoosY** [comment]
> @JackBoosY Do you still insist not to separate this port?

Why do we need to separate them?

**Comment by talregev** [comment]
> > @JackBoosY Do you still insist not to separate this port?
> 
> Why do we need to separate them?

For the port that not connected to the errors and need to wait for other PRs, can be fixed and be on the master.

**Comment by JackBoosY** [comment]
> > > @JackBoosY Do you still insist not to separate this port?
> > 
> > 
> > Why do we need to separate them?
> 
> For the ports that not connected to the errors and not need to wait for other PRs, can be fixed and be on the master.

But they are done for fix.

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thank you!

**Pull request merged**